modules/ROOT/pages/partner-connector-upgrade.adoc

```diff
@@ -0,0 +1,406 @@
+mvn clean
+mkdir target 
+mvn verify \
+    -DruntimeProduct=MULE_EE \
+    -DruntimeVersion=$RUNTIME_VERSION \
+    -Dmunit.jvm=$MUNIT_JVM \
+    -Dmtf.javaopts="--illegal-access=deny" > ./target/test.log
+----
+
+After running your MTF tests, go to the `target/illegal-access.log` file and check for classes or dependencies that misbehave. 
+
+You can also use the following command to exclude the known warnings outside of your custom connector: 
+
+[source,bash]
+----
+cat target/illegal-access.log | sort | uniq | grep -Ev "org.mule.module.artifact|org.mule.metadata|org.mule.runtime|org.mule.service"
+----
+
+[[test-for-java-17-compatibility-directly]]
+==== Test for Java 17 Compatibility Directly
+
+Run your MTF tests to test compatibility of your custom connector against Java 17. 
+
+As mentioned previously, you can have a single build pipeline that runs against all supported Java versions. You can also choose to set up another temporary build pipeline for Java 17 so your main build pipeline doesn't become unstable. After you upgrade to Java 17, you can discard the temporary build pipeline and converge on your main build pipeline.
+
+. Set the path to your JVM installation in the MUNIT_JVM variable (you must install it yourself). You must also set JAVA_HOME to Java 8. 
+. Ensure the following MTF dependencies are set in your custom connector’s `pom.xml` file:
++
+* munit 3.1.0
+* munit-extensions-maven-plugin 1.2.0
+* mtf-tools 1.2.0
+* mule-maven-plugin 4.1.0
+* mule-extensions-maven-plugin 1.6.0-rc1
++
+. These MTF dependencies have a minimum Mule version of 4.3.0. To ensure your MTF tests do not validate against Mule runtime versions earlier than 4.3.0, add this to the munit-plugin configuration of your custom connector’s `pom.xml` file: 
++
+[source,xml,linenums]
+----
+<configuration>
+	[...]
+<runtimeConfiguration>
+    <discoverRuntimes>
+        <minMuleVersion>${minVersion}</minMuleVersion>
+        <includeSnapshots>false</includeSnapshots>
+        <product>EE</product>
+    </discoverRuntimes>
+</runtimeConfiguration>
+</configuration>
+----
+
+You can run MTF tests against Java 17 only with Mule runtime 4.6.0 and later. For Mule runtime 4.5.x and earlier, you can run MTF tests only against Java 8 and Java 11. 
+
+MUnit 3.1 is compatible only with Mule runtime 4.3.0 and later. If your connector is compatible with Mule runtime 4.2.0 and earlier, you must create a legacy profile that overrides the MUnit version.
+
+Use the following bash script to test your custom connector against Java 17: 
+
+[source,bash]
+----
+#!/bin/bash
+RUNTIME_VERSION=4.6.0
+MUNIT_JVM=/Library/Java/JavaVirtualMachines/temurin-17.jdk/Contents/Home/bin/java
+mvn clean
+mkdir target
+mvn verify \
+   -DruntimeProduct=MULE_EE \
+   -DruntimeVersion=$RUNTIME_VERSION \
+   -Dmunit.jvm=$MUNIT_JVM \
+   -Dmule.module.tweaking.validation.skip=true \
+   -Dmule.jvm.version.extension.enforcement=LOOSE > ./target/test.log
+----
+
+=== Read Your Tests 
```

Comment:
Is this part of testing or parallel to testing? If part, should it be H4?

Reply:
It is parallel so H3